### PR TITLE
Add description annotation for scala case classes

### PR DESCRIPTION
**Pull request by narape**
# TL;DR
Add description annotation for scala case classes

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
* A new scala annotation was created. 
* The `SdkScalaType` search for it when reading the case classes fields
* When building the variable map it will set up the description or empty String if nothing was found

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/3303

## Follow-up issue
_NA_

### CODE REVIEW

**Review by sonjaer — APPROVED** [review]

**Review by andresgomezfrr — COMMENTED** [review]
Late, but ✅
